boards/litex_arty/README.md

```diff
@@ -0,0 +1,137 @@
+LiteX SoC on the Digilent Arty-A7 FPGA Board
+============================================
+
+This board is targeting a SoC bitstream built using
+[LiteX](https://github.com/enjoy-digital/litex), for the [Digilent
+Arty-A7 FPGA
+board](https://reference.digilentinc.com/reference/programmable-logic/arty-a7/start).
+
+Since LiteX is a SoC builder, the individual generated bitstreams can
+differ significantly depending on the LiteX release and configuration
+options used. This board definition currently targets and has been
+tested with
+- [the LiteX SoC generator, revision
+  444a605dea](https://github.com/enjoy-digital/litex/tree/444a605deae6a561dbe2c49bf3062eae6f3cd887)
+- using the included [target
+  file](https://github.com/enjoy-digital/litex/blob/444a605deae6a561dbe2c49bf3062eae6f3cd887/litex/boards/targets/arty.py)
+- built around a VexRiscv-CPU
+- along with the folling configuration options:
+
+  ```
+  --uart-baudrate=1000000
+  --cpu-variant=full
+  --csr-data-width=32
+  --timer-uptime
+  --integrated-rom-size=0xb000
+  --with-ethernet
+  ```
+
+Many bitstream customizations can be represented in the Tock board by
+simply changing the variables in `src/litex_generated.rs`. To support
+a different set of FPGA cores and perform further modifications, the
+`src/main.rs` file will have to be modified.
+
+
+Warning
+-------
+
+This board is still in development. The memory protection (PMP)
+mechanism is not yet integrated into the VexRiscv core and more
+peripherals require drivers. Nonetheless, the Tock kernel works and
+can run multiple userspace applications.
+
+The following on-board components and cores are supported:
+- [X] Timer (with uptime support)
+- [X] UART output via USB-FTDI
+- [X] Green LEDs
+- [X] 100MBit/s Ethernet MAC
+
+The following components and cores require porting:
+- [ ] Compressed instruction support in the VexRiscv CPU
```

Comment:
This is a hardware limitation right? Do we need to list that here?

Reply:
Right, I'll put it on my local todo list instead. VexRiscv _does_ support compressed instruction support (which would allow us to use the `riscv32imc` target, hence have hardware multiplication support), though it's not enabled in the `Full`-build which LiteX ships. I'll create a PR with them.